teuthology/task/ceph_ansible.py

```diff
@@ -61,25 +79,82 @@ class CephAnsible(ansible.Ansible):
     )
 
     def __init__(self, ctx, config):
-        config = config or dict()
+        super(CephAnsible, self).__init__(ctx, config)
+        config = self.config or dict()
         if 'playbook' not in config:
-            config['playbook'] = self._default_playbook
+            self.playbook = self._default_playbook
+        else:
+            self.playbook = self.config['playbook']
         if 'repo' not in config:
-            config['repo'] = os.path.join(teuth_config.ceph_git_base_url,
-                                          'ceph-ansible.git')
-        super(CephAnsible, self).__init__(ctx, config)
+            self.config['repo'] = os.path.join(teuth_config.ceph_git_base_url,
+                                               'ceph-ansible.git')
+        # default vars to dev builds
+        if 'vars' not in config:
+            vars = dict()
+            config['vars'] = vars
+        vars = config['vars']
+        if 'ceph_dev' not in vars:
+            vars['ceph_dev'] = True
+        if 'ceph_dev_key' not in vars:
+            vars['ceph_dev_key'] = 'https://download.ceph.com/keys/autobuild.asc'
+        if 'ceph_dev_branch' not in vars:
+            vars['ceph_dev_branch'] = ctx.config.get('branch', 'master')
 
-    def get_inventory(self):
+    def setup(self):
+        super(CephAnsible, self).setup()
+        # generate hosts file based on test config
+        self.generate_hosts_file()
+        # use default or user provided playbook file
+        pb_buffer = StringIO()
+        pb_buffer.write('---\n')
+        yaml.safe_dump(self.playbook, pb_buffer)
+        pb_buffer.seek(0)
+        playbook_file = NamedTemporaryFile(
+            prefix="ceph_ansible_playbook_",
+            dir='/tmp/',
+            delete=False,
+        )
+        playbook_file.write(pb_buffer.read())
+        playbook_file.flush()
+        self.playbook_file = playbook_file.name
+
+    def execute_playbook(self):
         """
-        Stub this method so we always generate the hosts file
+        Execute ansible-playbook
+
+        :param _logfile: Use this file-like object instead of a LoggerFile for
+                         testing
         """
-        pass
+
+        # everything from vars in config go into extra-vars
+        # TODO add option to directly copy them to group_vars/all
+        extra_vars = dict()
+        extra_vars.update(self.config.get('vars', dict()))
+        gvar = yaml.dump(extra_vars, default_flow_style=False)
+        self.gvars = self._write_hosts_file(prefix='teuth_ansible_gvar',
+                                            content=gvar)
```

Comment:
I guess I don't want to block on this, but I really dislike using unclear abbreviations like `gvars` instead of something clear, like `extra_vars_file`